doc/adr/0013-error-responses.md

```diff
@@ -0,0 +1,115 @@
+# 13. Error responses
+
+Date: 202x-xx-xx
+
+## Status
+
+Proposal
+
+## Context
+
+We want to show error messages in the frontend. For this the backend needs to provide more information to the frontend
+than the status code of the response.
+
+It is also possible that multiple validation errors occur at the same time. In that case we want to be able to send all
+of them to the frontend and include which part of the request the validation error is applicable to.
+
+The purpose of this ADR is similar to these ADRs of the caselaw project:
+
+- https://github.com/digitalservicebund/ris-backend-service/blob/main/doc/adr/0014-error-handling-concept.md
+- https://github.com/digitalservicebund/ris-backend-service/blob/main/doc/adr/0017-addressing-of-validation-errors.md
+
+[RFC-9457 (Problem Details for HTTP APIs)](https://www.rfc-editor.org/rfc/rfc9457) addresses how such information can be
+sent by an API.
+
+## Decision
+
+We follow [RFC-9457 (Problem Details for HTTP APIs)](https://www.rfc-editor.org/rfc/rfc9457) for our error responses.
+All error responses are JSON objects.
+
+The `type` field always starts with `/errors/`. Therefore, all errors are relative to our application.
+
+For errors related to a specific norm the `instance` field is filled with the eli of the norm starting with `/eli/`.
+
+Both `title` and `details` are written in english.
+
+### Multiple error messages in one response
+
+For errors that can have multiple messages (e.g. after a validation) we mostly follow the example from Section 3 of
+RFC-9457 for `HTTP/1.1 422 Unprocessable Content`.
+
+The multiple errors MUST be of the same general type. Otherwise, only the most severe error is reported. E.g. it is
+possible to send multiple `norm-not-valid` errors but not a `norm-not-valid` error at the same time as
+a `norm-not-found` error.
+
+The response includes an additional `errors` array. The elements of the `errors` array can have the same fields as the
+global object. If no `type` is provided it is assumed that it is the same as the parent type.
+
+For both JSON and LegalDocML.de some additional fields are defined to reference parts of a JSON-object or LDML.de
+document:
+
+#### JSON
+
+- `pointer`: a JSON-Pointer ([RFC-6901](https://www.rfc-editor.org/info/rfc6901)) referencing the field of the JSON
+  object responsible for this error
+
+#### LegalDocML.de
+
+- `eid`: an eId (LegalDocML.de) referencing the part of the norm responsible for the error (preferred)
+- `guid`: a GUID (LegalDocML.de) referencing the part of the norm responsible for the error (optionally or as
+  alternative to the eid)
+
+### Displaying messages for the user
+
+To create messages to display to the user the `type` must be used in combination with the extension members. The `title`
```

Comment:
I guess the frontend would then use the `type` which will probably contain parameters matching the extension members to build the message with those right?

Reply:
yes exactly thats the idea